src/purchase/views/purchase_view.py

```diff
@@ -77,6 +78,17 @@ def create(self, request):
         with transaction.atomic():
             user = User.objects.select_for_update().get(id=user.id)
 
+            cached_client_id = cache.get(f"purchase_client_id_{client_id}")
+            if cached_client_id:
+                return Response(
+                    {
+                        "detail": "This purchase has already been processed",
+                    },
+                    status=400,
+                )
+
+            cache.set(f"purchase_client_id_{client_id}", True, timeout=60)
```

Comment:
Nit: It might be better to implement idempotent behavior and return the original transaction with a 2xx series response code.